packages/next/server/next-server.ts

```diff
@@ -1130,11 +1130,11 @@ export default class NextNodeServer extends BaseServer {
           const rewritePath = result.response.headers.get(
             'x-middleware-rewrite'
           )!
-          const { newUrl, parsedDestination } = prepareDestination({
-            appendParamsToQuery: false,
-            destination: rewritePath,
-            params: _params,
+          const parsedDestination = parseUrl(rewritePath)
+          const newUrl = formatUrl({
+            ...parsedDestination,
             query: {},
+            search: '',
           })
```

Comment:
```suggestion
          const newUrl = parsedDestination.pathname
```

It looks like `newUrl` should contain only pathname?